glibc: align enable-kernel with current

curtdept opened the pull request:
This allows for a more targetted glibc to be built without legacy code. 

Some additional info @ https://www.gnu.org/software/libc/manual/html_node/Configuring-and-compiling.html

```Signed-off-by: Curtis Deptuck <curtdept@me.com>```

curtdept commented:
I'm on the fence able this, should we just drive it with $(LINUX_VERSION) to prevent constant maintenance.

Most of the legacy code this is effecting atm is with old SYSV calls and clock calls (incl. workaround for y3k in 5.1+)

dedeckeh commented:
To keep the maintenance burden to a minimum it would be better if this is updated automatically if the kernel is updated

curtdept commented:
> To keep the maintenance burden to a minimum it would be better if this is updated automatically if the kernel is updated

Ya, think aligning it with the build version would make sense, will update.

curtdept committed:
glibc: align enable-kernel with current

This allows for a more targetted glibc to be built without legacy code.

Some additional info @ https://www.gnu.org/software/libc/manual/html_node/Configuring-and-compiling.html

Signed-off-by: Curtis Deptuck <curtdept@me.com>

curtdept commented:
@dedeckeh updated and build/run tested, looks better then I though tbh.

hauke commented:
This is the minimum version.
If we have kernel 5.4 and 5.10 in our tree and the toolchain gets build with kernel 5.10, glibc will probably not work with the 5.4 target.
I would continue to use there the lowest kernel version we support.

curtdept commented:
> This is the minimum version.
> If we have kernel 5.4 and 5.10 in our tree and the toolchain gets build with kernel 5.10, glibc will probably not work with the 5.4 target.
> I would continue to use there the lowest kernel version we support.

It would be dynamic now based on selected kernel. So I think we satisfy that.

dedeckeh commented:
@hauke Do I understand your comments

> This is the minimum version.
> If we have kernel 5.4 and 5.10 in our tree and the toolchain gets build with kernel 5.10, glibc will probably not work with the 5.4 target.
> I would continue to use there the lowest kernel version we support.

@Hauke Do I understand it correctly you're not in favor of doing the automatic update as different kernel versions can be in use by different targets ?

curtdept commented:
Accidentally delete branch haha :)

hauke commented:
I would prefer if we don this manually. This is the minimum kernel version the compiled glibc should support. There is no problem when we use the glibc on a more recent kernel version, but it will not work on an older one.

curtdept commented:
Closing as timed out, just going to use it in my private branch where it works well.
